fxtest/lifecycle_test.go

```diff
@@ -117,6 +118,108 @@ func TestLifecycle(t *testing.T) {
 	})
 }
 
+func TestEnforceTimeout(t *testing.T) {
+	// These tests directly call Start and Stop
+	// rather than RequireStart and RequireStop
+	// because EnforceTimeout does not apply to those.
+
+	t.Run("StartHookTimeout", func(t *testing.T) {
+		t.Parallel()
+
+		wait := make(chan struct{}, 1)
+		defer close(wait)
+
+		spy := newTB()
```

Comment:
Should we verify that the test was not marked as failed?

```suggestion
		spy := newTB()
		defer func() { assert.Zero(t, spy.failures) }()
```